### PR TITLE
Dark Mode for the status site

### PR DESCRIPTION
**Pull request by brandong512**
- [X] Cookies Implemented
- [X] Sass theming system built
- [X] Toggle button at the top (day/night) according to spec

@akshay1502 
@rohan09-raj 
@whyDontI 

dark mode for status site

Not sure why the long chain of commits, but I'm open to learning more about it/correcting it

### CODE REVIEW

**Review by rohan09-raj — COMMENTED** [review]
@brandong512 Please check why the tests/checks are failing by clicking on the `details`
image

File: src/pages/index.tsx [hunk 1]
```diff
@@ -114,6 +124,8 @@ const Index: FC = () => {
   }, [isLoading, response]);
 
   useEffect(() => {
+    console.log("in use effect: ", checkThemeHistory(document.cookie, query))
```
Comment (shubham-y) [inline src/pages/index.tsx:127]: Please avoid adding console statements to production

**Review by shubham-y — CHANGES_REQUESTED** [review]
Good work 😃
Requesting some changes

File: src/pages/index.tsx (hunk 1 above)
Comment (brandong512) [reply src/pages/index.tsx:127]: No problem I'll go ahead and remove that

File: src/components/PullRequestList/index.tsx [hunk 2]
```diff
@@ -95,9 +100,14 @@ const PullRequestList: FC<PullRequestListProps> = ({ prType }) => {
     window.addEventListener('scroll', onScroll);
     return () => window.removeEventListener('scroll', onScroll);
   }, []);
+  const [mainDarkMode, setMainDarkMode] = useState(getDefaultOrTransferDark(query))
 
+  const themeSetter = () => {
+    document.cookie = setCookie(!mainDarkMode);
+    setMainDarkMode(!mainDarkMode);
+  }
```
[the remark this answers is not in the document]
Comment (brandong512) [reply src/components/PullRequestList/index.tsx:108]: I was actually struggling to do something like this. The issue here is that `document.cookie` is a global argument that relies on the component itself loading. If I try to put this argument in a helper function and it's called immediately before the DOM loads, then an error will show saying `document doesn't exist`.

Any suggestions?

**Comment by shubham-y** [comment]
@brandong512 Please add the dark mode feature changes for [Availability panel ](https://status.realdevsquad.com/availability-panel) also which is currently behind a feature flag.

**Review by shubham-y — COMMENTED** [review]
The background color of toastify modal does not change after switching from dark mode to light mode
image

**Review by shubham-y — COMMENTED** [review]
Suggestion: Add a calendar icon with a white border for dark mode
image

**Comment by brandong512** [comment]
Sounds good I'll go ahead and start working on these changes. Sorry for the delay!

**Comment by brandong512** [comment]
- [X] added dark mode to availability panel
- [X] added dark mode suggestion for calendar icon 
- [X] fixed dark mode for toast

@akshay1502
@rohan09-raj
@whyDontI

**Pull request closed without merging**